tests/cli/test_plot.py

```diff
@@ -152,3 +162,13 @@ def test_plot_dissociation_curve_print_table_output_file(run_cli_command, genera
     result = run_cli_command(plot.cmd_plot_dissociation_curve, options)
     assert f'Success: Table saved to {output_file}' in result.output
     assert Path.exists(output_file)
+
+
+@pytest.mark.usefixtures('aiida_profile')
+def test_plot_dissociation_curve_wrong_workchain(run_cli_command, generate_eos_node):
+    """Test the `plot_dissociation_curve` command in case the provided work chain is incorrect."""
+    node = generate_eos_node().store()
+
+    options = [str(node.pk), '--print-table']
+    with pytest.raises(AssertionError):
```

Comment:
As said in the other command, I would have this raise a `ValueError` instead and this should actually be caught in the plot command and then turned into a `echo_critical`. Having the command except is not the nicest I think

Reply:
Actually, this one is testing if the command fails with `echo_critical` when the wrong type of work chain is provided, e.g. for the `EquationOfStateWorkChain`:

```console
$ aiida-common-workflows plot eos -t 10362
Critical: node WorkChainNode<10362> does not correspond to an EquationOfStateWorkChain.
```

I first wanted to check if this message was in the output:

```Python
@pytest.mark.usefixtures('aiida_profile')
def test_plot_dissociation_curve_wrong_workchain(run_cli_command, generate_eos_node):
    """Test the `plot_dissociation_curve` command in case the provided work chain is incorrect."""
    node = generate_eos_node().store()

    options = [str(node.pk), '--print-table']
    result = run_cli_command(plot.cmd_plot_dissociation_curve, options)
    assert 'does not correspond to a DissociationCurveWorkChain' in result.output
```

However, this leads to an `AssertionError` when running `pytest`:

```Python
E           AssertionError: Traceback (most recent call last):
E               File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/click/testing.py", line 329, in invoke
E                 cli.main(args=args or (), prog_name=prog_name, **extra)
E               File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/click/core.py", line 782, in main
E                 rv = self.invoke(ctx)
E               File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
E                 return ctx.invoke(self.callback, **ctx.params)
E               File "/home/mbercx/.virtualenvs/aiida-dev/lib/python3.8/site-packages/click/core.py", line 610, in invoke
E                 return callback(*args, **kwargs)
E               File "/home/mbercx/envs/aiida-dev/code/aiida-common-workflows/aiida_common_workflows/cli/plot.py", line 104, in cmd_plot_dissociation_curve
E                 echo.echo_critical(
E               File "/home/mbercx/envs/aiida-dev/code/aiida-core/aiida/cmdline/utils/echo.py", line 127, in echo_critical
E                 sys.exit(ExitCode.CRITICAL)
E             SystemExit: ExitCode.CRITICAL
E             
E           assert SystemExit(<ExitCode.CRITICAL: 1>) is None
E            +  where SystemExit(<ExitCode.CRITICAL: 1>) = <Result SystemExit(<ExitCode.CRITICAL: 1>)>.exception
```

So I decided to check if this `AssertionError` was raised instead. Not sure if there is a better approach?

Reply:
```
result = run_cli_command(plot.cmd_plot_dissociation_curve, options, raises=SystemExit)
```
should work